packages/atomic/src/components/common/interface/analytics-config.ts

```diff
@@ -55,17 +55,18 @@ export function getNextAnalyticsConfig(
   searchEngineConfig: EngineConfiguration,
   enabled: boolean
 ): AnalyticsConfiguration {
-  const defaultConfiguration: AnalyticsConfiguration = {
+  const configuration: AnalyticsConfiguration = {
     enabled,
     documentLocation: document.location.href,
     ...(document.referrer && {originLevel3: document.referrer}),
   };
 
-  if (searchEngineConfig.analytics) {
-    return {
-      ...defaultConfiguration,
-      ...searchEngineConfig.analytics,
-    };
-  }
-  return defaultConfiguration;
+  const analyticsConfiguration = searchEngineConfig.analytics ?? {};
+  Object.assign(
+    analyticsConfiguration,
+    augmentAnalyticsConfigWithAtomicVersion()
+  );
+  Object.assign(configuration, searchEngineConfig.analytics);
```

Comment:
why object.assign and no spread operator ?

Reply:
Taste mostly. To use the spread operator here, I would need to have something like:
```
{
  ...defaultConfiguration,
  analytics: {
     ...analyticsConfiguration,
     ...augmentAnalyticsConfigWithAtomicVersion()
  }
}
```
I think it's less readable than the Object.assign